torchvision/transforms/v2/_augment.py

```diff
@@ -135,3 +138,150 @@ def _transform(
             inpt = F.erase(inpt, **params, inplace=self.inplace)
 
         return inpt
+
+
+class _BaseMixupCutmix(Transform):
+    def __init__(self, *, alpha: float, num_categories: Optional[int] = None, labels_getter="default") -> None:
+        super().__init__()
+        self.alpha = alpha
+        self._dist = torch.distributions.Beta(torch.tensor([alpha]), torch.tensor([alpha]))
+
+        self.num_categories = num_categories
+
+        self.labels_getter = labels_getter
+        self._labels_getter = _parse_labels_getter(labels_getter)
+
+    def forward(self, *inputs):
+        inputs = inputs if len(inputs) > 1 else inputs[0]
+        flat_inputs, spec = tree_flatten(inputs)
+        needs_transform_list = self._needs_transform_list(flat_inputs)
+
+        if has_any(flat_inputs, PIL.Image.Image, datapoints.BoundingBox, datapoints.Mask):
+            raise TypeError(f"{type(self).__name__}() does not support PIL images, bounding boxes and masks.")
+
+        labels = self._labels_getter(inputs)
+        if labels is None:
+            msg = "Couldn't find a label in the inputs."
+            if self.labels_getter == "default":
+                msg = f"{msg} To overwrite the default find behavior, pass a callable for labels_getter."
+            raise RuntimeError(msg)
+        elif not isinstance(labels, torch.Tensor):
+            raise ValueError(f"The labels must be a torch.Tensor, but got {type(labels)} instead.")
+        elif labels.ndim in {1, 2}:
+            if labels.ndim == 2 and self.num_categories is not None and labels.shape[-1] != self.num_categories:
+                raise ValueError(
+                    f"2D labels are assumed to be probability based, "
```

Comment:
What does it mean to be probability based? We probably just mean "one-hot-encoded" for the input (ground-truth) labels right?

Reply:
We had some major discussions about this before you joined. The problem with saying "one-hot encoded labels" is that after Cutmix or Mixup that is no longer true. One-hot encoding basically means turning `3` into `[0, 0, 0, 1, 0]` given we have 4 categories. However, due to the nature of Mixup and Cutmix, afterwards the label could look like `[0, 0.5, 0, 0.5, 0]` if an image with label `1` was mixed with one of label `3`. And that can no longer be called "one-hot encoding".

To find better suiting terms, I looked at some loss functions, e.g. [`CrossEntropyLoss`](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html?highlight=crossentropyloss#torch.nn.CrossEntropyLoss):

> Target: If containing **class indices**, shape ()(), (N)(N) or (N,d1,d2,...,dK)(N,d1​,d2​,...,dK​) with K≥1K≥1 in the case of K-dimensional loss where each value should be between [0,C)[0,C). If containing **class probabilities**, same shape as the input and each value should be between [0,1][0,1].

(emphasis mine)

And this is why I'm using the terms "index based" and "category based" here.

Reply:
I understand the ambiguity for labels *after* they're transformed by CutMix/MixUp, but when we're talking about the input (i.e. before being transformed), the labels tensor should always be one-hot-encoded (or just ints) in the stricter sense, right?

Since this error message is only referring to the input, I feel like we can go ahead and use "one-hot-encoded" safely?

Reply:
> the labels tensor should always be one-hot-encoded

I guess that's not the case, typically when one combines `Compose([CutMix(), MixUp()])`. The transforms seem to work fine even when "dirichlet-encoded" inputs are passed in, which seem to be by design?